File: 1800. Maximum Ascending Subarray Sum/maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        curr = nums[0]
        res = curr

        for idx in range(1, len(nums)):
            if nums[idx-1] < nums[idx]:
                curr += nums[idx]
            else:
                curr = nums[idx]

            res = max(res, curr)

        return res

File: 1800. Maximum Ascending Subarray Sum/maximum-ascending-subarray-sum.go
package main

func maxAscendingSum(nums []int) int {
    curr := nums[0]
    res := curr

    for idx := 1; idx < len(nums); idx++ {
        if nums[idx-1] < nums[idx] {
            curr += nums[idx]
        } else {
            curr = nums[idx]
        }
        
        if res < curr {
            res = curr
        }
    }
    return res

}